### PR TITLE
Implement listMethods for lua generic app

### PR DESCRIPTION
**Pull request by eddyashton**
Our `listMethods` RPC previously just returned the names of all registered handlers. Since the generic lua app (and presumably future similar generic interpreters) use the _default_ endpoint, they don't actually report the supported methods (ie - they don't include `LOG_record` for the lua logging app). This fixes that.

While doing this I also fiddled with the `test_requirements` to generate parameterised variants, so we can easily say "this test requires exactly these methods", rather than looking for a specific app. I also added the `ensure_reqs` meta-decorator to remove some of the boilerplate (ie - to consistently skip if `enforce_reqs` is False). I'm sure there's some way to reduce the boilerplate even further, but I don't see it straight away.

### CODE REVIEW

File: tests/suite/test_requirements.py [hunk 1]
```diff
@@ -19,61 +19,72 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-# TODO: Parameterise this decorator once we add a test that requires a
-# different number of nodes
-def at_least_2_nodes(func):
+def ensure_reqs(check_reqs, func):
```
Comment (jumaffre) [inline tests/suite/test_requirements.py]: Nice!

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**